src/Illuminate/Pagination/AbstractPaginator.php

```diff
@@ -646,152 +625,35 @@ public static function useBootstrapFive()
      *
-     * @return bool
+     * @param  string|null  $view
+     * @param  array  $data
+     * @return \Illuminate\Contracts\Support\Htmlable
      */
-    public function isNotEmpty()
+    public function render($view = null, $data = [])
     {
-        return $this->items->isNotEmpty();
+        return static::viewFactory()->make($view ?: static::$defaultSimpleView, array_merge($data, [
+            'paginator' => $this,
+        ]));
     }
 
     /**
-     * Get the number of items for the current page.
-     *
-     * @return int
-     */
-    public function count(): int
-    {
-        return $this->items->count();
-    }
-
-    /**
-     * Get the paginator's underlying collection.
-     *
-     * @return \Illuminate\Support\Collection
-     */
-    public function getCollection()
-    {
-        return $this->items;
-    }
-
-    /**
-     * Set the paginator's underlying collection.
-     *
-     * @param  \Illuminate\Support\Collection  $collection
-     * @return $this
-     */
-    public function setCollection(Collection $collection)
-    {
-        $this->items = $collection;
-
-        return $this;
-    }
-
-    /**
-     * Get the paginator options.
+     * Get the instance as an array.
      *
      * @return array
      */
-    public function getOptions()
-    {
-        return $this->options;
-    }
-
-    /**
-     * Determine if the given item exists.
-     *
-     * @param  mixed  $key
-     * @return bool
-     */
-    public function offsetExists($key): bool
-    {
-        return $this->items->has($key);
-    }
-
-    /**
-     * Get the item at the given offset.
-     *
-     * @param  mixed  $key
-     * @return mixed
-     */
-    public function offsetGet($key): mixed
-    {
-        return $this->items->get($key);
-    }
-
-    /**
-     * Set the item at the given offset.
-     *
-     * @param  mixed  $key
-     * @param  mixed  $value
-     * @return void
-     */
-    public function offsetSet($key, $value): void
-    {
-        $this->items->put($key, $value);
-    }
-
-    /**
-     * Unset the item at the given key.
-     *
-     * @param  mixed  $key
-     * @return void
-     */
-    public function offsetUnset($key): void
-    {
-        $this->items->forget($key);
-    }
-
-    /**
-     * Render the contents of the paginator to HTML.
-     *
-     * @return string
-     */
-    public function toHtml()
-    {
-        return (string) $this->render();
-    }
-
-    /**
-     * Make dynamic calls into the collection.
-     *
-     * @param  string  $method
-     * @param  array  $parameters
-     * @return mixed
-     */
-    public function __call($method, $parameters)
-    {
-        return $this->forwardCallTo($this->getCollection(), $method, $parameters);
-    }
-
-    /**
-     * Render the contents of the paginator when casting to a string.
-     *
-     * @return string
-     */
-    public function __toString()
-    {
-        return (string) $this->render();
+    public function toArray()
```

Comment:
The `Paginator` and the `LengthAwarePaginator` have some matching attributes. So I moved up the `toArray` method to their parent class with their common array attributes. So from the child classes it's enough just add their own data.